pkg/network/tracer/tracer_linux_test.go

```diff
@@ -2571,6 +2576,201 @@ func setupDropTrafficRule(tb testing.TB) (ns string) {
+				require.NoError(t, tlsConn.Handshake())
+
+				return port, scenario
+			},
+			validation: func(t *testing.T, tr *Tracer, port uint16, scenario uint16) {
+				require.Eventuallyf(t, func() bool {
+					return validateTLSTags(t, tr, port, scenario)
+				}, 3*time.Second, 100*time.Millisecond, "couldn't find TLS connection matching: dst port %v", port)
+			},
+		})
+	}
+	tests = append(tests, tlsTest{
+		name: "Invalid-TLS-Handshake",
+		postTracerSetup: func(t *testing.T) (uint16, uint16) {
+			// server that accepts connections but does not perform TLS handshake
+			listener, err := net.Listen("tcp", "localhost:0")
+			require.NoError(t, err)
+			t.Cleanup(func() { listener.Close() })
+
+			go func() {
+				for {
+					conn, err := listener.Accept()
+					if err != nil {
+						return
+					}
+					go func(c net.Conn) {
+						defer c.Close()
+						buf := make([]byte, 1024)
+						_, _ = c.Read(buf)
+						// Do nothing with the data
+					}(conn)
+				}
+			}()
+
+			// Retrieve the actual port from the listener address
+			addr := listener.Addr().String()
+			_, portStr, err := net.SplitHostPort(addr)
+			require.NoError(t, err)
+			portInt, err := strconv.Atoi(portStr)
+			require.NoError(t, err)
+			port := uint16(portInt)
+
+			// Client connects to the server
+			conn, err := net.Dial("tcp", addr)
+			require.NoError(t, err)
+			defer conn.Close()
+
+			// Send invalid TLS handshake data
+			_, err = conn.Write([]byte("invalid TLS data"))
+			require.NoError(t, err)
+
+			// Since this is invalid TLS, scenario can be set to something irrelevant, e.g., TLS.VersionTLS12
+			// or just 0 since the validation doesn't rely on the scenario for this test.
+			return port, tls.VersionTLS12
+		},
+		validation: func(t *testing.T, tr *Tracer, port uint16, _ uint16) {
+			// Verify that no TLS tags are set for this connection
+			require.Eventually(t, func() bool {
+				payload := getConnections(t, tr)
+				for _, c := range payload.Conns {
+					if c.DPort == port && c.ProtocolStack.Contains(protocols.TLS) {
+						t.Log("Unexpected TLS protocol detected for invalid handshake")
+						return false
+					}
+				}
+				return true
+			}, 3*time.Second, 100*time.Millisecond)
+		},
+	})
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if ebpftest.GetBuildMode() == ebpftest.Fentry {
+				t.Skip("protocol classification not supported for fentry tracer")
+			}
+			t.Cleanup(func() { tr.RemoveClient(clientID) })
+			t.Cleanup(func() { _ = tr.Pause() })
+
+			tr.RemoveClient(clientID)
+			require.NoError(t, tr.RegisterClient(clientID))
+			require.NoError(t, tr.Resume(), "enable probes - before post tracer")
+			port, scenario := tt.postTracerSetup(t)
+			require.NoError(t, tr.Pause(), "disable probes - after post tracer")
+			tt.validation(t, tr, port, scenario)
+		})
+	}
+}
+
+func validateTLSTags(t *testing.T, tr *Tracer, port uint16, scenario uint16) bool {
+	payload := getConnections(t, tr)
```

Comment:
`getConnections` uses `require.NoError` inside of it, and you call `validateTLSTags(t,...)` in a ` require.Eventually` method. Any failure in `getConnections` will terminate immediately your test, regardless of the timeout configured in `require.Eventually`

Reply:
I believe this issue exists in most NPM tests as we call getConnections inside of `require.Eventually`. I will fix this in a follow up PR